mlrun/serving/routers.py

```diff
@@ -244,7 +254,150 @@ class OperationTypes(str, Enum):
     explain = "explain"
 
 
-class VotingEnsemble(BaseModelRouter):
+class _ParallelRunInterface:
+    def __init__(
+        self, executor_type: Union[ParallelRunnerModes, ExecutorTypes, str], **kwargs
+    ):
+        """
+        Interface for execute parallel runners
+
+        :param executor_type: The desired way for execute parallel run.
+                              Have 3 option :
+                              * array - running one by one
+                              * process - running in separated process
+                              * thread - running in separated threads
+        """
+        if isinstance(executor_type, ExecutorTypes):
+            executor_type = str(executor_type)
+            logger.warn(
+                "ExecutorTypes is deprecated and will be removed in 1.5.0, use ParallelRunnerModes instead",
+                # TODO: In 1.5.0 do changes
+                PendingDeprecationWarning,
+            )
+        self.executor_type = ParallelRunnerModes(executor_type)
+        self._pool: Union[
+            concurrent.futures.ProcessPoolExecutor,
+            concurrent.futures.ThreadPoolExecutor,
+        ] = None
+        super().__init__(**kwargs)
+
+    def _init_pool(
+        self,
+    ) -> Union[
+        concurrent.futures.ProcessPoolExecutor, concurrent.futures.ThreadPoolExecutor
+    ]:
+        """
+
+        Get the tasks pool of this runner. If the pool is `None`,
+        a new pool will be initialized according to `executor_type`.
+
+        :return: The tasks pool
+        """
+        if self._pool is None:
+            if self.executor_type == ParallelRunnerModes.process:
+                # init the context and route on the worker side (cannot be pickeled)
+                server = self.context.server.to_dict()
+                routes = {}
+                for key, route in self.routes.items():
```

Comment:
I couldn't find where `self.routes` are being initialized, please elaborate as it is not very clear. As `_ParallelRunInterface` doesn't inherit from any class I am not sure where the kwargs are being passed to

Reply:
Add if statement in `__init__`  method